Library/Homebrew/os/linux/elf.rb

```diff
@@ -78,6 +79,24 @@ def dynamic_elf?
     end
   end
 
+  def corrupted_elf?
+    # elf files that throw the following message when parsed with readelf:
+    # Warning: possibly corrupt ELF header
+    # it has a non-zero program header offset, but no program headers
+    # Example: the early-section-headers.so unittest file from the rust package
+    offset = 0
+    nbr_program_headers = 0
+    Utils.popen_read("readelf", "-h", to_path).each_line do |line|
+      if line["Start of program headers:"]
+        offset = line.scan(/\d+/)[0].to_i
+      end
+      if line["Number of program headers:"]
+        nbr_program_headers = line.scan(/\d+/)[0].to_i
+      end
+    end
+    nbr_program_headers.zero? && !offset.zero?
```

Comment:
Would it be better to just check stderr for `Warning: possibly corrupt ELF header - it has a non-zero program header offset, but no program headers` since it seems like this code is duplicating the check `readelf` is already doing?

Reply:
I like this solution. That's how I handled `patchelf` failing in some cases.
https://github.com/Homebrew/brew/blob/6ae600b5e67008e81ef2e3d595ebfb986912e0a7/Library/Homebrew/extend/os/linux/keg_relocate.rb#L21-L25